### PR TITLE
Apply ReplaceAddCounter with ReplaceMoved for ETB Counter

### LINKED ISSUE #2481: Compleated PW can't gain their full strength from counter doublers
https://www.reddit.com/r/askajudge/comments/tg0k5i/tamiyo_compleated_sage_and_doubling_season/

Apparently doubling them can be done before compleated.

@Hanmac 
seems like a very ugly thing for us
Compleated might have to be changed from Moved to AddCounter?

### LINKED ISSUE #3939: RE execution needs better lookahead
Judges say Devouring a counter replacer should still modify the ETB counters:

https://www.reddit.com/r/mtgrules/comments/q41up2/winding_constrictor_corpsejack_menace_and_devour/

### PR DESCRIPTION
**Pull request by Hanmac**
Closes #2481
Closes #3939

we might need more checks around these negative values

might remove the NoteCounters? it is now only used with static?

### CODE REVIEW

File: forge-game/src/main/java/forge/game/card/CardFactoryUtil.java [hunk 1]
```diff
@@ -2208,8 +2208,8 @@ public static void addReplacementEffect(final KeywordInterface inst, final CardS
 
             inst.addReplacement(re);
         } else if (keyword.equals("Compleated")) {
-            String sb = "etbCounter:LOYALTY:PhySpent:CheckSVar$ PhySpent | SVarCompare$ LT0:This planeswalker" +
-                    " enters with two fewer loyalty counters for each Phyrexian mana symbol life was paid for";
+            String sb = "etbCounter:LOYALTY:PhySpent:CheckSVar$ PhySpent | SVarCompare$ LT0:Compleated (This planeswalker" +
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/card/CardFactoryUtil.java:2211]: get the reminder text from the instance?

**Comment by tool4ever** [comment]
Compleated Tamiyo + Doubling Season crashes:```

Game-0 > java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because "effectOnly" is null
	at forge.game.replacement.ReplaceAddCounter.canReplace(ReplaceAddCounter.java:38)
	at forge.game.replacement.ReplacementHandler$1.visit(ReplacementHandler.java:146)
	at forge.game.replacement.ReplacementHandler$1.visit(ReplacementHandler.java:1)
	at forge.util.Visitor.visitAll(Visitor.java:16)
	at forge.game.Game.forEachCardInGame(Game.java:737)
	at forge.game.Game.forEachCardInGame(Game.java:723)
	at forge.game.replacement.ReplacementHandler.getReplacementList(ReplacementHandler.java:119)
	at forge.game.replacement.ReplacementHandler.run(ReplacementHandler.java:210)
	at forge.game.replacement.ReplacementHandler.run(ReplacementHandler.java:200)
	at forge.game.GameAction.changeZone(GameAction.java:360)
	at forge.game.GameAction.moveTo(GameAction.java:804)
	at forge.game.GameAction.moveTo(GameAction.java:762)
	at forge.game.GameAction.moveToPlay(GameAction.java:897)
	at forge.game.ability.effects.PermanentEffect.resolve(PermanentEffect.java:31)
	at forge.game.ability.SpellApiBased.resolve(SpellApiBased.java:51)
	at forge.game.ability.AbilityUtils.resolveApiAbility(AbilityUtils.java:1415)
	at forge.game.ability.AbilityUtils.resolve(AbilityUtils.java:1353)
	at forge.game.zone.MagicStack.resolveStack(MagicStack.java:607)
```

File: forge-game/src/main/java/forge/game/card/CardFactoryUtil.java (hunk 1 above)
Comment (Hanmac) [reply forge-game/src/main/java/forge/game/card/CardFactoryUtil.java]: done

**Comment by Hanmac** [comment]
> Compleated Tamiyo + Doubling Season crashes:
> 
> ```
> Game-0 > java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because "effectOnly" is null
> 	at forge.game.replacement.ReplaceAddCounter.canReplace(ReplaceAddCounter.java:38)
> 	at forge.game.replacement.ReplacementHandler$1.visit(ReplacementHandler.java:146)
> 	at forge.game.replacement.ReplacementHandler$1.visit(ReplacementHandler.java:1)
> 	at forge.util.Visitor.visitAll(Visitor.java:16)
> 	at forge.game.Game.forEachCardInGame(Game.java:737)
> 	at forge.game.Game.forEachCardInGame(Game.java:723)
> 	at forge.game.replacement.ReplacementHandler.getReplacementList(ReplacementHandler.java:119)
> 	at forge.game.replacement.ReplacementHandler.run(ReplacementHandler.java:210)
> 	at forge.game.replacement.ReplacementHandler.run(ReplacementHandler.java:200)
> 	at forge.game.GameAction.changeZone(GameAction.java:360)
> 	at forge.game.GameAction.moveTo(GameAction.java:804)
> 	at forge.game.GameAction.moveTo(GameAction.java:762)
> 	at forge.game.GameAction.moveToPlay(GameAction.java:897)
> 	at forge.game.ability.effects.PermanentEffect.resolve(PermanentEffect.java:31)
> 	at forge.game.ability.SpellApiBased.resolve(SpellApiBased.java:51)
> 	at forge.game.ability.AbilityUtils.resolveApiAbility(AbilityUtils.java:1415)
> 	at forge.game.ability.AbilityUtils.resolve(AbilityUtils.java:1353)
> 	at forge.game.zone.MagicStack.resolveStack(MagicStack.java:607)
> ```

fixed

**Comment by tool4ever** [comment]
I tested _Mycoloth_ devouring _Winding Constrictor_ = 3 counters
so it seems #3939 is also fixed 👍

**Comment by Hanmac** [comment]
@tool4ever @Agetian is this MR ready to go?

i the AI might need a bit more tweaking when choosing replacement effects, but i think this is good to go?

**Comment by Agetian** [comment]
Yeah, looks good to me so far :+1:

**Comment by tool4ever** [comment]
because of the complexity I haven't tested it too deeply yet

**Comment by Hanmac** [comment]
@tool4ever I would like to see this merged before a possible Bloomborrow release (because it is important for a Combo Deck)

**Review by tool4ever — APPROVED** [review]
well I don't want to block it
what doesn't work yet is the CantPutCounter + simultaneous ETB case mentioned in the issue, but a new one for that might be better

**Comment by Hanmac** [comment]
yeah, we can fix CantPutCounter later again

**Pull request merged**